zcash_client_sqlite/src/lib.rs

```diff
@@ -320,6 +328,19 @@ impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletRead for W
         );
     }
 
+    fn get_unspent_transparent_output(
+        &self,
+        _outpoint: &OutPoint,
+    ) -> Result<Option<WalletTransparentOutput>, Self::Error> {
+        #[cfg(feature = "transparent-inputs")]
+        return wallet::get_unspent_transparent_output(self.conn.borrow(), _outpoint);
+
+        #[cfg(not(feature = "transparent-inputs"))]
+        panic!(
+            "The wallet must be compiled with the transparent-inputs feature to use this method."
+        );
```

Comment:
This seems unfortunate. Why can't the existence of the method be conditional?

Reply:
Trait methods are not supposed to have their presence or absence determined by feature flags. The correct thing will be to factor this out into the `TransparentInputSource` type introduced by #1003.

Reply:
Actually, it can't be conditional, because bounds can't vary based on feature flags.